llms/mlx_lm/models/base.py

```diff
@@ -2,6 +2,47 @@
 from dataclasses import dataclass
 
 import mlx.core as mx
+import mlx.nn as nn
+
+
+class DynamicNTKScalingRoPE(nn.Module):
```

Comment:
Can you move this to the internlm2 file? I don't think it's used anywhere else so let's keep it there for now to keep the base file clean.

Reply:
Done ✅

Reply:
I had put it there because a lot transformers models including llama use it to scale context window.

- Llama 2 and 3: https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L269
- Gemma: https://github.com/huggingface/transformers/blob/a1a34657d41627b21dddf2bf9cc55941329a60b6/src/transformers/models/gemma/modeling_gemma.py#L137
- StableLM: https://github.com/huggingface/transformers/blob/a1a34657d41627b21dddf2bf9cc55941329a60b6/src/transformers/models/stablelm/modeling_stablelm.py#L120
- Falcon: https://github.com/huggingface/transformers/blob/a1a34657d41627b21dddf2bf9cc55941329a60b6/src/transformers/models/falcon/modeling_falcon.py#L172
- Olmo: https://github.com/huggingface/transformers/blob/a1a34657d41627b21dddf2bf9cc55941329a60b6/src/transformers/models/olmo/modeling_olmo.py#L119
- gpt neo: https://github.com/huggingface/transformers/blob/a1a34657d41627b21dddf2bf9cc55941329a60b6/src/transformers/models/gpt_neox/modeling_gpt_neox.py#L541
- Phi: https://github.com/huggingface/transformers/blob/a1a34657d41627b21dddf2bf9cc55941329a60b6/src/transformers/models/phi/modeling_phi.py#L120